### ISSUE TITLE
RC heex template compiler problem doesn't indicate the file

### ISSUE THREAD

**Issue by brainlid** [phoenixframework/phoenix#4453]
### Environment

* Elixir version (elixir -v): Elixir 1.12.2 (compiled with Erlang/OTP 24)
* Phoenix version (mix deps): `{:phoenix, "~> 1.6.0-rc.0", override: true}`
* Operating system: Arch Linux

### Expected behavior

Compiler (or heex compiler) problems would tell me the file with the problem.

### Actual behavior

- On an umbrella application where the "web" interface is an app called "Web".
- Using Phoenix RC version and migrating a project to it, I have some errors but I can't tell where the error is. 

```
== Compilation error in file lib/web/views/layout_view.ex ==
** (Phoenix.LiveView.HTMLTokenizer.ParseError) nofile:21:56: expected closing `"` for attribute value
    (phoenix_live_view 0.16.0) lib/phoenix_live_view/html_tokenizer.ex:360: Phoenix.LiveView.HTMLTokenizer.handle_attr_value_double_quote/6
    (phoenix_live_view 0.16.0) lib/phoenix_live_view/html_engine.ex:90: Phoenix.LiveView.HTMLEngine.handle_text/3
    (eex 1.12.2) lib/eex/compiler.ex:48: EEx.Compiler.generate_buffer/4
    (eex 1.12.2) lib/eex/compiler.ex:80: EEx.Compiler.generate_buffer/4
    (phoenix_view 1.0.0) lib/phoenix/template.ex:371: Phoenix.Template.compile/3
    (phoenix_view 1.0.0) lib/phoenix/template.ex:165: anonymous fn/4 in Phoenix.Template."MACRO-__before_compile__"/2
    (elixir 1.12.2) lib/enum.ex:2385: Enum."-reduce/3-lists^foldl/2-0-"/3
    (phoenix_view 1.0.0) expanding macro: Phoenix.Template.__before_compile__/1
    lib/web/views/layout_view.ex:1: Web.LayoutView (module)
    (elixir 1.12.2) lib/kernel/parallel_compiler.ex:319: anonymous fn/4 in Kernel.ParallelCompiler.spawn_workers/7
```

The contents of `layout_view.ex` are:

```elixir
defmodule Web.LayoutView do
  use Web, :view
end
```

**Comment by marciol** [comment]
Hi @brainlid, I tried to reproduce those errors without success. 

I created a [repo](https://github.com/marciol/hello_umbrella_phoenix_issue_4453) with my attempt to reproduce the issue.

**Comment by brainlid** [comment]
Thank you @marciol for creating a starter project! I [created a PR](https://github.com/marciol/hello_umbrella_phoenix_issue_4453/pull/1) against that project that re-creates the problem. Also, from your working example I realized how I could solve it for my specific project as well.

I hope the demonstration of the problem is still helpful!

**Comment by petermm** [comment]
should be fixed in https://github.com/phoenixframework/phoenix_live_view/issues/1584 - but maybe the error message needs to be expanded to include a string interpolation example?

currently its:
```

    message = """
    expected closing `>` or `/>`
    This may happen if there is an EEx interpolation inside a tag,
    which is not supported. Instead of
        <a href="<%= @url %>">Text</a>
    do
        <a href={@url}>Text</a>
    """
```

so maybe add something like this: (for improved dx)

```
or from
<div class="small <%= @conditional_class %>">text</div>
to
<div class="small #{@conditional_class}">text</div>
```

**Comment by 123mitnik** [comment]
the same error happens when trying to use conditional rendering, the old eex way to interpolation worked
```
<div <%= if info = get_flash(@conn, :info) do %> x-data="{ isOpen: true }" <%else %> x-data="{ isOpen: false }" <% end %>
...
>
```
but now in HEEx it gives

```expected closing `>` or `/>```

how we must rewrite it ?

**Comment by 123mitnik** [comment]
i've putted it outside the element .. but still interested can we use if..do..else inside the tag (i.e. how to rewrite the above)

**Comment by petermm** [comment]
for alpine/x-data I think it looks something like this:
<p x-data={"{msg: 'Alpine works: #{if true, do: "yes", else: "false"}!'}"} x-text="msg" >Alpine does not work :(</p>

so for your code:
<div x-data={"{isOpen: #{if get_flash(@conn, :info), do: true, else: false}}"} >div content</div>

**Comment by josevalim** [comment]
This is fixed on LiveView 0.16.1 (and better error messages coming on 0.16.2).

**Comment by brainlid** [comment]
Yes! Verified it was a problem in 0.16.0, was fixed in 0.16.1, and better errors in 0.16.2.
